### DISCUSSION TITLE
DevOpsTools in Cyfrin Foundry Course Broken by Recent Foundry Security Update - Seeking Solutions

### DISCUSSION THREAD

**Discussion by LordAdhaar** [Cyfrin/foundry-full-course-cu#2572] [General]
I'm currently working through the Cyfrin Updraft Foundry course and have encountered an issue that I believe stems from recent Foundry security updates. I'm hoping to get some clarity and possibly find a solution or alternative approach.

The Issue:

When trying to run the FundFundMe script from the course using the command:
forge script script/Interactions.s.sol:FundFundMe --rpc-url $(ANVIL_RPC_URL) --broadcast --interactives 1

I'm encountering the following error:
Copyscript failed: vm.readDir: the path broadcast is not allowed to be accessed for read operations

Root Cause:

After some investigation, it appears that this error is due to Foundry implementing new security measures that restrict access to certain directories, including the broadcast directory, when running scripts. This seems to be affecting the functionality of the DevOpsTools library used in the course, specifically the get_most_recent_deployment() function which relies on reading from the broadcast directory.

Current Workaround:
I've managed to work around this issue by manually inputting the contract address instead of relying on DevOpsTools to fetch it. However, this solution feels like a step backward in terms of automation and convenience.

Questions:

Are you aware of this compatibility issue between the course materials and recent Foundry updates?
Do you have any recommended updates or modifications for the DevOpsTools library to make it compatible with these new security measures?

**Comment by EngrPips** [comment]
Hello @LordAdhaar, It should work fine if you put the below in your `Foundry.toml`

```js
fs_permissions = [{ access = "read", path = "./broadcast" }, {access = "read", path = "./Images/"}]
```

**Reply by LordAdhaar** [reply]
Bro tf, worked like a charm. 
How are you so active here all the time and how do you know so much?
Teach me ser.

**Reply by EngrPips** [reply]
I have been working as a Teacher Assistant for Patrick since March of this year and have been reading every single discussion since then. Even if I didn't end up providing the solution to the issue, I read the provided solution and recognize that problem the next time I encounter it. Plus, I do heavy personal studying to stay relevant. So, if you consciously stay around long enough, you will inevitably know a lot. Keep crushing it, friend.

**Reply by LordAdhaar** [reply]
Sometimes the more I try understand things, the more I get confused I get lol. Do you know any good discord groups apart from the Cyfrin Updraft discord to join to connect with likeminded people?

Also, thanks for the advice ser.

**Reply by EngrPips** [reply]
That's good because the more confused you get, the clearer it is actually getting, but you don't know. You only got confused because your brain is trying to connect the dot, and breaking out of that confusion means you totally understand it. I do more book reading than community joining, so honestly I am only active on Cyfrin Discord, and I don't know another discord,

**Reply by LordAdhaar** [reply]
Which books your reading?
Let me know some.
